.github/workflows/smoke-tests.yml

```diff
@@ -0,0 +1,47 @@
+# Workflow for running smoke tests for pull requests etc.
+
+name: Pre merge tests
+
+on:
+  pull_request:
+
+  workflow_dispatch:
+
+jobs:
+  pre-merge-test:
+    name: Pre merge tests
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      # yarn / node modules
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: cache yarn
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          # note: uses a separate (dev) cache, different from the build workflow
+          key: ${{ runner.os }}-yarn-dev-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-dev-
+
+      # inject npmrc
```

Comment:
I think these above the "- name" comments is a bit unnecessary as it is basically repeated in the "- name" yaml property.